src/Validation/Validator/AbstractNumberValidator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ScrumWorks\OpenApiSchema\Validation\Validator;
+
+use ScrumWorks\OpenApiSchema\Validation\BreadCrumbPath;
+use ScrumWorks\OpenApiSchema\Validation\Result\ValidationResultBuilderInterface;
+
+abstract class AbstractNumberValidator extends AbstractValidator
+{
+    protected function validateNumberConstrains(
+        ValidationResultBuilderInterface $resultBuilder,
+        ?float $data,
+        ?float $min,
+        ?float $max,
+        ?bool $exclusiveMin,
+        ?bool $exclusiveMax,
+        ?float $multipleOf,
+        BreadCrumbPath $breadCrumbPath
+    ): void {
+        $exclusiveMin ??= false;
+        $exclusiveMax ??= false;
+
+        if ($min !== null && $exclusiveMin && $min >= $data) {
+            $resultBuilder->addExclusiveMinimumViolation($min, $breadCrumbPath);
+        }
+        if ($min !== null && ! $exclusiveMin && $min > $data) {
+            $resultBuilder->addMinimumViolation($min, $breadCrumbPath);
+        }
+
+        if ($max !== null && $exclusiveMax && $max <= $data) {
+            $resultBuilder->addExclusiveMaximumViolation($max, $breadCrumbPath);
+        }
+        if ($max !== null && ! $exclusiveMax && $max < $data) {
+            $resultBuilder->addMaximumViolation($max, $breadCrumbPath);
+        }
+
+        if ($multipleOf !== null && $data % $multipleOf !== 0) {
```

Comment:
pokud to pouzijes na float mas tak trochu problem, jelikoz `0.9 % 0.3` vyhodi `Fatal error: Uncaught DivisionByZeroError: Modulo by zero`, musis tedy pouzit `fmod`, jenze opet bacha `fmod(0.9, 0.3)` mi vratilo `float(5.5511151231258E-17)` - takze tam bude potreba nejake lehke zaokrouhlovani :)